File: Quera/1-Programming-basics/easy/008-Prime-Numbers/main.c
#include <stdio.h>

int main()
{
    //Variables
    int a=0,b=0;
    int i,j;
    int isPrime[10001] = {0};


    //Input
    scanf("%d %d",&a,&b);



    //Calculation
    for (i = 2; i < 10001; i++)
    {
        isPrime[i] = 1;
    }


    for (i = 2; i <= b; i++)
    {
        if (isPrime[i])
        {
            for (j = i*i; j <= b; j +=i)
            {
                isPrime[j] = 0;
            }
        }
    }

    //Output
    for (i = a; i <= b; i++)
    {
        if (isPrime[i]) printf("%d\n",i);
    }

    return 0;
}

File: Quera/1-Programming-basics/easy/008-Prime-Numbers/Main.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {

        //Variables
        int a,b;
        boolean[] isPrime = new boolean[10001];


        //Input
        Scanner getScanner = new Scanner(System.in);
        a = getScanner.nextInt();
        b = getScanner.nextInt();


        //Calculation
        for (int i = 2; i < 10001; i++) {
            isPrime[i] = true;
        }

        for (int i = 2; i < 10001; i++) {
            if (isPrime[i]) {
                for (int j = i*i; j <= b; j+=i) {
                    isPrime[j] = false;
                }
            }
        }


        //Output
        for (int i = a; i <= b; i++) {
            if (isPrime[i]) {
                System.out.println(i);
            }
        }
    }
}
